src/Microsoft.DotNet.VersionTools/Automation/GitHubVersionsRepoUpdater.cs

```diff
@@ -148,21 +148,39 @@ public GitHubVersionsRepoUpdater(GitHubAuth gitHubAuth, GitHubProject project)
                     }
                     catch (HttpRequestException ex)
                     {
-                        int nextTry = i + 1;
-                        if (nextTry < MaxTries)
+                        bool retrySucceeded = await TryRetry(i, ex.Message);
```

Comment:
This method and the variable are really confusing... after reading it all I can see what it's doing, but these names don't describe it with it as far as I can tell. I see it's reducing duplication--I suggest replacing this loop with this util that handles it to clean it all up:

https://github.com/dotnet/arcade/blob/master/src/Common/ExponentialRetry.cs

(Nit: The method is also missing `Async` suffix.)